satellite/satellitedb/certdb.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type certDB struct {
+	db *dbx.DB
+}
+
+func (certs *certDB) Set(ctx context.Context, nodeID storj.NodeID, pi *identity.PeerIdentity) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	tx, err := certs.db.Begin()
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+		} else {
+			err = errs.Combine(err, tx.Rollback())
+		}
+	}()
+
+	if pi == nil {
+		return Error.New("Peer Identity cannot be nil")
+	}
+	chain := identity.EncodePeerIdentity(pi)
+
+	var id []byte
+	query := `SELECT node_id FROM peer_identities WHERE serial_number = ?;`
+	err = tx.QueryRow(certs.db.Rebind(query), pi.Leaf.SerialNumber.Bytes()).Scan(&id)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// when storagenode can get a new leaf certificate, which should be store as a new entry. But currently
+			// for a storagenode to get a new leaf identity, it is controlled in our alpha network. When this changes, then same
+			// storagenode can have multiple entries, which is handled here
+			_, err = tx.Exec(certs.db.Rebind(`INSERT INTO peer_identities ( serial_number, peer_identity, node_id, update_at ) VALUES ( ?, ?, ?, ? );`), pi.Leaf.SerialNumber.Bytes(), chain, nodeID.Bytes(), time.Now())
+			if err != nil {
+				return Error.Wrap(err)
+			}
+			return nil
+		}
+		return Error.Wrap(err)
+	}
+
+	// already public key exists, just return
+	return nil
+}
+
+// Get gets the public key based on the certificate's nodeID
+func (certs *certDB) Get(ctx context.Context, nodeID storj.NodeID) (_ *identity.PeerIdentity, err error) {
+	defer mon.Task()(&ctx)(&err)
+	dbxPeerID, err := certs.db.Get_PeerIdentity_By_NodeId_OrderBy_Desc_UpdateAt(ctx, dbx.PeerIdentity_NodeId(nodeID.Bytes()))
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	if dbxPeerID == nil {
+		return nil, Error.New("unknown nodeID :%+v: %+v", nodeID.Bytes(), err)
+	}
+
+	peer, err := identity.DecodePeerIdentity(ctx, dbxPeerID.PeerIdentity)
+	return peer, Error.Wrap(err)
+}
+
+// BatchGet gets the public key based on the certificate's serial number
+func (certs *certDB) BatchGet(ctx context.Context, nodeIDs []storj.NodeID) (peers []*identity.PeerIdentity, err error) {
```

Comment:
maybe `peerIdents` or something instead of `peers` as `peer` means something else and is used in other, unrelated parts of the code.